manifests/gem.pp

```diff
@@ -69,17 +69,22 @@
     fail('You must declare a ruby_version for rbenv::gem')
   }
 
+  $environment_for_install = concat(["RBENV_ROOT=${install_dir}"], $env)
+  $version_for_exec_name = regsubst($version, '[^0-9]+', '_', 'EG')
+
+  # Using shell if and substitution, build a test if rubygems is version 3 or
+  # greater.
+  $gem_ver3_test_command_prefix = 'IFS=. read -a gem_version_components <<< "$(gem --version)"; if [ "${gem_version_components[0]}" -ge 3 ]; then'
+  $base_gem_install_command = "gem install ${gem} --version '${version}' --source '${source}'"
+
   if ($skip_docs) {
-    $docs = '--no-ri --no-rdoc'
+    $gem_install_command = "/bin/true ; ${gem_ver3_test_command_prefix} ${base_gem_install_command} --no-document; else ${base_gem_install_command} --no-ri --no-rdoc; fi"
   } else {
-    $docs = ''
+    $gem_install_command = $base_gem_install_command
   }
 
-  $environment_for_install = concat(["RBENV_ROOT=${install_dir}"], $env)
-  $version_for_exec_name = regsubst($version, '[^0-9]+', '_', 'EG')
```

Comment:
We need these two variables still.

Reply:
These two variables are still there, I just moved them above the logic used to determine the command that would be used for gem.  My reasoning here was that it seemed more readable to keep the calculations that would be used to determine the command closer to the exec.  I don't have strong feelings on this if you want to move the variables back.